packages/flutter_tools/lib/src/windows/migrations/build_architecture_migration.dart

```diff
@@ -0,0 +1,117 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import '../../base/file_system.dart';
+import '../../base/project_migrator.dart';
+import '../../cmake_project.dart';
+import 'utils.dart';
+
+const String _cmakeFileToolBackendBefore = r'''
+add_custom_command(
+  OUTPUT ${FLUTTER_LIBRARY} ${FLUTTER_LIBRARY_HEADERS}
+    ${CPP_WRAPPER_SOURCES_CORE} ${CPP_WRAPPER_SOURCES_PLUGIN}
+    ${CPP_WRAPPER_SOURCES_APP}
+    ${PHONY_OUTPUT}
+  COMMAND ${CMAKE_COMMAND} -E env
+    ${FLUTTER_TOOL_ENVIRONMENT}
+    "${FLUTTER_ROOT}/packages/flutter_tools/bin/tool_backend.bat"
+      windows-x64 $<CONFIG>
+  VERBATIM
+)
+''';
+
+const String _cmakeFileToolBackendAfter = r'''
+add_custom_command(
+  OUTPUT ${FLUTTER_LIBRARY} ${FLUTTER_LIBRARY_HEADERS}
+    ${CPP_WRAPPER_SOURCES_CORE} ${CPP_WRAPPER_SOURCES_PLUGIN}
+    ${CPP_WRAPPER_SOURCES_APP}
+    ${PHONY_OUTPUT}
+  COMMAND ${CMAKE_COMMAND} -E env
+    ${FLUTTER_TOOL_ENVIRONMENT}
+    "${FLUTTER_ROOT}/packages/flutter_tools/bin/tool_backend.bat"
+      ${FLUTTER_TARGET_PLATFORM} $<CONFIG>
+  VERBATIM
+)
+''';
+
+const String _cmakeFileTargetPlatformBefore = r'''
+# TODO: Move the rest of this into files in ephemeral. See
+# https://github.com/flutter/flutter/issues/57146.
+set(WRAPPER_ROOT "${EPHEMERAL_DIR}/cpp_client_wrapper")
+
+# === Flutter Library ===
+''';
+
+const String _cmakeFileTargetPlatformAfter = r'''
+# TODO: Move the rest of this into files in ephemeral. See
+# https://github.com/flutter/flutter/issues/57146.
+set(WRAPPER_ROOT "${EPHEMERAL_DIR}/cpp_client_wrapper")
+
+# Set fallback configurations for older versions of the flutter tool.
+if (NOT DEFINED FLUTTER_TARGET_PLATFORM)
+  set(FLUTTER_TARGET_PLATFORM "windows-x64")
+endif()
+
+# === Flutter Library ===
+''';
+
+/// Migrates Windows build to target specific architecture.
+/// In more, it deletes old runner folder
+class BuildArchitectureMigration extends ProjectMigrator {
+  BuildArchitectureMigration(
+    WindowsProject project,
+    Directory buildDirectory,
+    super.logger
+  )
+    : _cmakeFile = project.managedCmakeFile,
+      _buildDirectory = buildDirectory;
+
+  final File _cmakeFile;
+  final Directory _buildDirectory;
+
+  @override
+  void migrate() {
+    final Directory oldRunnerDirectory = _buildDirectory
+      .parent
+      .childDirectory('runner');
+    if (oldRunnerDirectory.existsSync()) {
+      logger.printTrace('''
+Delete previous build folder ${oldRunnerDirectory.path}.
+New binaries can be found in ${_buildDirectory.childDirectory('runner').path}.
+''');
+      oldRunnerDirectory.deleteSync(recursive: true);
```

Comment:
Let's still catch the exception here. The previous build directory contains executables the user might have run, so it's not unlikely this will throw:

```suggestion
      try {
        oldRunnerDirectory.deleteSync(recursive: true);
      } on FileSystemException catch (error) {
        final String path = oldRunnerDirectory.path;
        globals.printError(
          'Failed to remove $path. '
          'A program may still be using a file in the directory or the directory itself. '
          'To find and stop such a program, see: '
          'https://superuser.com/questions/1333118/cant-delete-empty-folder-because-it-is-used'
        );
      }
```